test/integration/report_test.rb

```diff
@@ -591,7 +605,7 @@ def setup
     Timecop.freeze(Time.utc(2010, 5, 12, 13, 33)) do
       post '/transactions.xml',
         :provider_key => @provider_key,
-        :transactions => {0 => {:app_id => @application.id, :usage => {'hits' => 1}, :timestamp => '2012'}}
+        :transactions => {0 => {:app_id => @application.id, :usage => {'hits' => 1}, :timestamp => '2012garbage2012'}}
```

Comment:
Can you adapt the comment belong accordingly?

Reply:
Mean below? :+1:

Reply:
The comment and even the test are also utterly wrong... `2012` would mean `20120101`, not `20121001`... :woman_facepalming:

Reply:
The comment looks OK. It says that because `2012` is invalid, the keys increased correspond to the current timestamp (defined by the Timecop call above) and not by the timestamp specified `2012`. Am I missing something?

Reply:
It is testing the negative case as well: that `20121001`, October 2012, monthly granularity, is not incremented. If 2012 _was_ understood, then it would have been January 2012.

Reply:
2012 is invalid. That's why Jan 2012 it's not increased.

Reply:
Yes, I'm saying it's a _negative_ test case: it's testing that October 2012 is not increased. For that matter, it could have been testing that November 1990, or February 9001, wouldn't be increased, because it has nothing to do with what's being fed to the test.

Reply:
IOW, it could have used year 1234567890 and it would still be wrong, because if that year was being tested, then why choose October? If anything, specifying just a year is done with month 01.

Reply:
I didn't see that `20121001`. It should have been `20120101` :joy: